lightning_client.go

```diff
@@ -696,6 +701,128 @@ type NetworkInfo struct {
 	NumZombieChans uint64
 }
 
+// AcceptorRequest contains the details of an incoming channel that has been
+// proposed to our node.
+type AcceptorRequest struct {
+	// NodePubkey is the pubkey of the node that wishes to open an inbound
+	// channel.
+	NodePubkey route.Vertex
+
+	// ChainHash is the hash of the genesis block of the relevant chain.
+	ChainHash []byte
+
+	// PendingChanID is the pending channel ID for the channel.
+	PendingChanID [32]byte
+
+	// FundingAmt is the total funding amount.
+	FundingAmt btcutil.Amount
+
+	// PushAmt is the amount pushed by the party pushing the channel.
+	PushAmt btcutil.Amount
+
+	// The dust limit of the initiator's commitment tx.
+	DustLimit btcutil.Amount
+
+	// MaxValueInFlight is the maximum msat amount that can be pending in the
+	// channel.
+	MaxValueInFlight btcutil.Amount
+
+	// ChannelReserve is the minimum amount of satoshis the initiator requires
+	// us to have at all times.
+	ChannelReserve btcutil.Amount
+
+	// MinHtlc is the smallest HTLC in millisatoshis that the initiator will
+	// accept.
+	MinHtlc lnwire.MilliSatoshi
+
+	// FeePerKw is the initial fee rate that the initiator suggests for both
+	// commitment transactions.
+	FeePerKw chainfee.SatPerKWeight
+
+	// CsvDelay is the number of blocks to use for the relative time lock in
+	// the pay-to-self output of both commitment transactions.
+	CsvDelay uint32
+
+	// MaxAcceptedHtlcs is the total number of incoming HTLC's that the
+	// initiator will accept.
+	MaxAcceptedHtlcs uint32
+
+	// ChannelFlags is a bit-field which the initiator uses to specify proposed
+	// channel behavior.
+	ChannelFlags uint32
+}
+
+func newAcceptorRequest(req *lnrpc.ChannelAcceptRequest) (*AcceptorRequest,
+	error) {
+
+	pk, err := route.NewVertexFromBytes(req.NodePubkey)
+	if err != nil {
+		return nil, err
+	}
+
+	var pending [32]byte
+	copy(pending[:], req.PendingChanId)
+
+	return &AcceptorRequest{
+		NodePubkey:       pk,
+		ChainHash:        req.ChainHash,
+		PendingChanID:    pending,
+		FundingAmt:       btcutil.Amount(req.FundingAmt),
+		PushAmt:          btcutil.Amount(req.PushAmt),
+		DustLimit:        btcutil.Amount(req.DustLimit),
+		MaxValueInFlight: btcutil.Amount(req.MaxValueInFlight),
+		ChannelReserve:   btcutil.Amount(req.ChannelReserve),
+		MinHtlc:          lnwire.MilliSatoshi(req.MinHtlc),
+		FeePerKw:         chainfee.SatPerKWeight(req.FeePerKw),
+		CsvDelay:         req.CsvDelay,
+		MaxAcceptedHtlcs: req.MaxAcceptedHtlcs,
+		ChannelFlags:     req.ChannelFlags,
+	}, nil
+}
+
+// AcceptorResponse contains the response to a channel acceptor request.
+type AcceptorResponse struct {
+	// Accept indicates whether to accept the channel.
+	Accept bool
+
+	// PendingChanId is the pending channel id to which this response
+	// applies.
+	PendingChanId []byte
```

Comment:
Do we need this field?

Reply:
Could set it directly from the request so that we don't need to set it!

Reply:
Yeah, mostly what I meant is that we pass the request containing `PendingChanId` to the closure and then return it in the response from the same closure. Not super important though. I'm also fine with having it in the response too.

Reply:
Decided to leave it in the request, because I can see some use case for wanting to log the pending channel id in the accept code, and can see us perhaps needing to come back to log it. Removed it from the response because we don't need it there. Sound ok?